clients/src/main/java/org/apache/kafka/common/requests/ListOffsetRequest.java

```diff
@@ -64,66 +73,117 @@ public PartitionData(long timestamp, int maxNumOffsets) {
         }
     }
 
+    /**
+     * Constructor for ListOffsetRequest v0
+     */
+    @Deprecated
     public ListOffsetRequest(Map<TopicPartition, PartitionData> offsetData) {
-        this(CONSUMER_REPLICA_ID, offsetData);
+        this(CONSUMER_REPLICA_ID, offsetData, 0);
     }
 
+    /**
+     * Constructor for ListOffsetRequest v0
+     */
+    @Deprecated
     public ListOffsetRequest(int replicaId, Map<TopicPartition, PartitionData> offsetData) {
-        super(new Struct(CURRENT_SCHEMA));
-        Map<String, Map<Integer, PartitionData>> topicsData = CollectionUtils.groupDataByTopic(offsetData);
+        this(replicaId, offsetData, 0);
+    }
+
+    /**
+     * Constructor for ListOffsetRequest v1.
+     */
+    public ListOffsetRequest(Map<TopicPartition, ?> targetTimes, int replicaId) {
```

Comment:
For followup: I don't think we came to a good resolution on this constructor, but having a wildcard parameter seems like a definite step backwards.